nncf/experimental/torch/fx/transformations.py

```diff
@@ -174,11 +207,16 @@ def constant_update_fn(model: torch.fx.GraphModule, node: torch.fx.Node, value:
 
     # Update metadata of the new constant node.
     previous_const = args[input_port_id]
-    new_constant.meta = copy(previous_const.meta)
-    new_constant.meta["val"] = value
+    consumer_nodes = list(
+        previous_const.users.keys()
+    )  # This list of consumer nodes will always be topologically sorted
+    # To ensure the updated node has the right order,
+    # we insert constant node before the node placed at the highest order in topological order.
+    with graph.inserting_before(consumer_nodes[0]):
+        new_constant = create_getattr_from_value(model, graph, node.name + "_updated_constant", value)
 
-    args[input_port_id] = new_constant
-    node.args = tuple(args)
+    previous_const.replace_all_uses_with(new_constant, propagate_meta=True)
+    graph.erase_node(previous_const)
```

Comment:
Why we need to call it explicitly? Looks line the next line `graph.eliminate_dead_code` is doing the same
```suggestion
```

Reply:
This was done as a test to ensure the node is deleted and so is the buffer. I will remove it now

Reply:
Done